opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -52,37 +56,54 @@
    *
    * @param <C> the carrier type, which also parametrizes the Format.
    * @param spanContext the SpanContext instance to inject into the carrier
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.inject() implementations must support io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
+   * @param format the Format of the carrier. All Tracer.inject() implementations
+   *               must support formats defined in io.opentracing.propagation.Format.Builtin
+   * @param carrier the carrier for the SpanContext state.
+   *
+   * @throws io.opentracing.propagation.UnsupportedFormatException when tracer does not support given format
    *
    * @see io.opentracing.propagation.Format
    * @see io.opentracing.propagation.Format.Builtin
+   * @see io.opentracing.propagation.UnsupportedFormatException
    */
   <C> void inject(SpanContext spanContext, Format<C> format, C carrier);
 
   /**
    * Extract a SpanContext from a `carrier` of a given type, presumably after propagation across a process boundary.
    *
+   * If tracer does not support given format, it must throw
+   * io.opentracing.propagation.UnsupportedFormatException so that instrumentation
+   * can try another format.
+   *
+   * If given carrier does not contain information to construct SpanContext,
+   * the tracer must return `null`. If SpanContext cannot be extracted because
+   * the data in the carrier is malformed, the tracer should return null, rather
+   * than throw an exception. The tracer may handle the exceptions internally
+   * by logging them or incrementing an error counter.
+   *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
    * TextMap httpHeadersCarrier = new AnHttpHeaderCarrier(httpRequest);
    * SpanContext spanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, httpHeadersCarrier);
-   * tracer.buildSpan('...').withChildOf(spanCtx).start();
+   * tracer.buildSpan('...').asChildOf(spanCtx).start();
    * }</pre>
    *
    * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in an
    * IllegalArgumentException.
    *
    * @param <C> the carrier type, which also parametrizes the Format.
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
-   *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
+   * @param format the Format of the carrier. All Tracer.extract() implementations
+   *               must support formats defined in io.opentracing.propagation.Format.Builtin
+   * @param carrier the carrier for the SpanContext state.
    *
-   * @return the SpanContext instance holding context to create a Span.
+   * @return the SpanContext instance extracted from the carrier holding context to create a Span or null if it cannot be extracted.
```

Comment:
> If Java supported multiple return values as Go does, this would've been simple 

The idiomatic way this happens in Java is either through a checked exception to indicate a recoverable error, a null object like the one I suggested, or a separate result object such as `ExtractResult` (assuming we are not targeting Java 7 and above; using `Optional` is a choice in that case). I'm leaning towards the null object pattern because it is cleanest in terms up API usage. If consensus is such that existing apps that downcast to implementations of a pre-1.0 spec can not be asked to change, then you can introduce `ExtractResult` or throw a checked exception. _Either_ of those options would be better than returning `null`.

> Tracer implementations need to check for it every time they are passed a span context. Users may also be affected.

I think requiring no-op behavior for null contexts from implementations is much preferable if it means orders of magnitude more code written by clients of the OT API have a simpler means of dealing with missing tracing contexts.

> So just to preserve the yes/no bit about the trace context in the request we'd be introducing this new concept with much greater impact.

You may say this is just semantics, but I think the concept is not new. It was always there but the API did not recognize it: _tracing is optional_; lack of a distributed tracing context should not cause failure in an app. I think the question is how we deal with it. 

As for apps that have a direct dependency on a particular tracing implementation; why do they care about the OT API in this case? Can you not keep using Jaeger API to extract tracing contexts? Alternatively, Jaeger is always free to return its own null context implementation from `extract` and check for it in addition to the one in the OT API. This limits the cost of having to preserve `SpanContext` downcasts to just Jaeger users.